File: app/src/main/java/vedder/xander/brewtracker/ui/adapter/RecipeAdapter.java
package vedder.xander.brewtracker.ui.adapter;

import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import vedder.xander.brewtracker.model.Recipe;
import vedder.xander.brewtracker.ui.view.Card;

public class RecipeAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    private List<Recipe> dataset;

    public RecipeAdapter(List<Recipe> dataset) {
        this.dataset = dataset;
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        Card card = new Card(parent.getContext());
        card.setLayoutParams(new ViewGroup.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT,
                ViewGroup.LayoutParams.WRAP_CONTENT
        ));
        return new RecipeViewHolder(card);
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
        ((RecipeViewHolder) holder).getCard().setRecipe(this.dataset.get(position));
    }

    @Override
    public int getItemCount() {
        return this.dataset.size();
    }

    public static class RecipeViewHolder extends RecyclerView.ViewHolder {

        private View card;

        public RecipeViewHolder(@NonNull View itemView) {
            super(itemView);
            this.card = itemView;
        }

        public Card getCard() {
            return (Card) card;
        }
    }
}

File: app/src/main/java/vedder/xander/brewtracker/recipe/ui/RecipeAdapter.kt
package vedder.xander.brewtracker.recipe.ui

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import vedder.xander.brewtracker.R
import vedder.xander.brewtracker.recipe.model.Recipe

class RecipeAdapter(
        private var dataset: MutableList<Recipe>
) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_card, parent, false)
        return RecipeViewHolder(view)
    }

    override fun getItemCount(): Int {
        return dataset.size
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        dataset[position].let { (holder as RecipeViewHolder).setRecipe(it) }
    }

    inner class RecipeViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        private val brewType: TextView = view.findViewById(R.id.card_chip)
        private val dateCreated: TextView = view.findViewById(R.id.card_date_created)
        private val title: TextView = view.findViewById(R.id.card_title)

        fun setRecipe(recipe: Recipe) {
            brewType.text = recipe.type.toString()
            title.text = recipe.name
            dateCreated.text = recipe.createdAt.toString()
        }
    }
}
